### PR TITLE
adding a JAVA client to test the home-recipes services

### PR DESCRIPTION
**Pull request by bhargavmehta01**

### CODE REVIEW

**Comment by kaushikchaubal** [comment]
I suggest to rename the branch from `app-client-java` to `test-client-java` to keep it consistent

File: client/app-client-java/src/main/proto/google/api/annotations.proto [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015, Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (kaushikchaubal) [inline client/app-client-java/src/main/proto/google/api/annotations.proto]: Why is there a proto folder inside the client? Your client should pick up the protos from the defs folder. Please delete this folder and use the protos from the defs folder which  has all the definitions already

**Review by kaushikchaubal — CHANGES_REQUESTED** [review]
Please update the two things  as mentioned in other comments:
1. Remove the duplicated proto folder and use the 'defs' folder which has all the proto definitions 
2. Rename the branch from app-client-java to test-client-java to keep it consistent

**Comment by bhargavmehta01** [comment]
- Removed the duplicated proto defs folder and now the project depends on the existing proto defs outside the project
- Renamed to 'test-client-java' for the consistency purpose

**Comment by kaushikchaubal** [comment]
thanks for your contribution! 💯

**Pull request merged**